pkg/workload/workload_test.go

```diff
@@ -643,3 +644,63 @@ func TestResourceUsage(t *testing.T) {
 		})
 	}
 }
+
+func TestAdmissionCheckStrategy(t *testing.T) {
+	cases := map[string]struct {
+		cq                  *kueue.ClusterQueue
+		wl                  *kueue.Workload
+		wantAdmissionChecks []string
+	}{
+		"AdmissionCheckStrategy with a flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(*utiltesting.MakeAdmissionCheckStrategy("ac1", "flavor1").Obj()).
+				Obj(),
+			wantAdmissionChecks: []string{"ac1"},
+		},
+		"AdmissionCheckStrategy without a flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(*utiltesting.MakeAdmissionCheckStrategy("ac1").Obj()).
+				Obj(),
+			wantAdmissionChecks: []string{"ac1"},
+		},
+		"Two AdmissionCheckStrategies, one with flavor, one without flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(
+					*utiltesting.MakeAdmissionCheckStrategy("ac1", "flavor1").Obj(),
+					*utiltesting.MakeAdmissionCheckStrategy("ac2").Obj()).
+				Obj(),
+			wantAdmissionChecks: []string{"ac1", "ac2"},
+		},
+		"Workload has no Admission": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(
+					*utiltesting.MakeAdmissionCheckStrategy("ac1", "flavor1").Obj(),
+					*utiltesting.MakeAdmissionCheckStrategy("ac2").Obj()).
+				Obj(),
+			wantAdmissionChecks: nil,
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			ctx := context.Background()
+			log := ctrl.LoggerFrom(ctx)
```

Comment:
Should we use our library?

https://github.com/kubernetes-sigs/kueue/blob/b9c346a83d965d7849e4e27cf7e06e5ee1ad22df/pkg/util/testing/context.go#L28

Reply:
Fixed